src/app.rs

```diff
@@ -585,65 +585,15 @@ impl<'a, 'b, T: State + 'a> ApplicationBuilder<'a, 'b, T> {
     ///
     /// # Errors
     ///
-    /// This method initializes the renderer, which uses a number of system APIs
-    /// that we cannot guarantee will function correctly on every system.
-    ///
-    /// Sources of possible error (none exhaustive):
+    /// This function creates systems and resources, which use any number of dependent
+    /// crates or APIs, which could result in any number of errors.
+    /// See each individual bundle for a description of the errors it could produce.
     ///
-    /// - Failure to create a window
-    /// - Failure to initialize the lower level graphics context (OpenGL, DX)
-    /// - Failure to initialize the thread pool
-    /// - Failure to compile shaders for the target platform
-    /// - linking error between shader variables and framework
-    ///
-    /// # Examples
-    ///
-    /// See the `renderable` example supplied in the examples folder for a fully
-    /// working example that illustrates how to configure the renderer.
-    /// The example below is a brief example showing how and where to get the
-    /// types needed to initialize the renderer.
-    ///
-    /// ~~~no_run
-    /// use amethyst::prelude::*;
-    /// use amethyst::renderer;
-    /// use amethyst::renderer::prelude::*;
-    /// 
-    /// struct NullState;
-    /// impl State for NullState {}
-    /// 
-    /// // Create a new display_config, we can tweak the configuration
-    /// // here, or load it from a file.
-    /// let mut display_config = renderer::Config::default();
-    /// display_config.title = "An Amazing Amethyst Game!".into();
-    /// display_config.dimensions = Some((640, 480));
-    ///
-    /// // A pipeline needs to be built to initialize the renderer
-    /// let pipeline_builder = Pipeline::build().with_stage(
-    ///     Stage::with_backbuffer()
-    ///         .clear_target([0.0, 0.0, 0.0, 1.0], 1.0)
-    ///         .with_model_pass(pass::DrawShaded::<PosNormTex>::new()),
-    /// );
-    ///     
-    /// // finally we are able to add the renderer to our application.
-    /// Application::build(NullState)
-    ///     .expect("Failed to initialize")
-    ///     .with_renderer(pipeline_builder, Some(display_config))
-    ///     .expect("Failed to create render system for application");
-    /// ~~~
-    pub fn with_renderer(
-        mut self,
-        pipeline: PipelineBuilder,
-        config: Option<DisplayConfig>,
-    ) -> Result<Self> {
-        use ecs::SystemExt;
-        use ecs::rendering::RenderSystem;
-        let render_sys = RenderSystem::build((&self.events, pipeline, config), &mut self.world)?;
-        Ok(
-            self.with_thread_local(render_sys)
-                .register_mesh_asset()
-                .register_texture_asset()
-                .register_material_not_yet_asset(),
-        )
+    pub fn with_bundle<B>(self, bundle: B) -> Result<Self>
```

Comment:
If a bundle offers multiple systems, how would one manage dependencies? Should maybe the documentation list the system names it adds?

Reply:
I'd put all the information that the user could need in the documentation for each bundle.